### PR TITLE
Update baselines + Logic change in reference checking

### PR DESCRIPTION
**Pull request by h-joo**
1. Update baselines which were missing.
2. Update reference (delete) when the files are missing for dte/tsc check.

### CODE REVIEW

File: src/harness/harnessIO.ts [hunk 1]
```diff
@@ -1063,7 +1063,12 @@ export namespace Compiler {
         tsSources: readonly TestFile[],
         prettyErrors: boolean | undefined,
         reason: string | undefined,
+        expectNull?: boolean,
     ) {
+        if (expectNull) {
+          // eslint-disable-next-line no-null/no-null
+          return Baseline.runBaseline(type + "/" + baselinePath.replace(/\.tsx?/, `.d.ts.diff`), /*actual*/ null);
+        }
```
Comment (dragomirtitian) [inline src/harness/harnessIO.ts:1071]: Why was this necessary ?

**Review by dragomirtitian — CHANGES_REQUESTED** [review]

File: src/harness/harnessIO.ts (hunk 1 above)
Comment (h-joo) [reply src/harness/harnessIO.ts:1071]: So we were polluting the repo by keeping files that aren't being tested against references (when tsc and dte results were different before thus checked-in to the repo but later when they became identical) and creating confusion because we weren't testing it, it took me some time to understand why the tests weren't complaining about the files that shouldn't be present.

**Review by dragomirtitian — APPROVED** [review]
